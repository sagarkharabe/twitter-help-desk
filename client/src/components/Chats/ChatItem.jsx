import React from "react";
import { observer } from "mobx-react";
import { appStore } from "../../store/appStore";
import { ListItem, Avatar } from "@material-ui/core";
import moment from "moment";

function ChatItem(props) {
  let { style, item } = props;
  const isUser = appStore.user.screen_name === item.user.screen_name;
  return (
    <div
      style={{
        marginLeft: isUser ? "35%" : "1%",
        display: "flex",
        flexDirection: "row"
      }}
    >
      <Avatar alt={item.user.name} src={item.user.profile_image_url} />
      <ListItem
        style={{
          marginLeft: "1%",
          marginBottom: "2%",
          width: "75%",
          borderWidth: "1px",
          borderStyle: "solid",
          borderColor: "#d3d3d3",
          borderRadius: "20px",
          borderTopLeftRadius: 0,
          backgroundColor: isUser ? "#d8edb8" : "#FBFBFB"
        }}
      >
        <span>
          <b className="user-name">{item.user.name}</b> <br />
          <p
            style={{
              fontSize: "1em",
              marginRight: "5px"
            }}
            className="user-tweet"
          >
            {item.text}
          </p>
          <p className="created-at" style={{ fontSize: "0.8em" }}>
            {moment(item.created_at).fromNow()}
          </p>
        </span>
      </ListItem>
    </div>
  );
}

export default observer(ChatItem);
