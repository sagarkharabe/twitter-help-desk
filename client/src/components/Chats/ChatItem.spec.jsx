import React from "react";
import { mount, shallow } from "enzyme";
import ChatItem from "./ChatItem";

const mockTweet = {
  user: {
    name: "test",
    profile_image_url: "string",
    created_at: "00:00:00",
    text: "some text"
  }
};

describe("Chat Item", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <ChatItem
        style={{
          marginLeft: "1%",
          backgroundColor: "#FBFBFB"
        }}
        item={mockTweet}
      />
    );
  });

  test("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("renders children", () => {
    expect(wrapper.find(".MuiAvatar-img")).toHaveLength(1);
    expect(wrapper.find(".user-name")).toHaveLength(1);
    expect(wrapper.find(".user-tweet")).toHaveLength(1);
    expect(wrapper.find(".created-at")).toHaveLength(1);
  });
});
