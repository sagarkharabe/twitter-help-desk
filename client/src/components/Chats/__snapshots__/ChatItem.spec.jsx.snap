// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Item renders component 1`] = `
<ChatItem
  item={
    Object {
      "user": Object {
        "created_at": "00:00:00",
        "name": "test",
        "profile_image_url": "string",
        "text": "some text",
      },
    }
  }
  style={
    Object {
      "backgroundColor": "#FBFBFB",
      "marginLeft": "1%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": "1%",
      }
    }
  >
    <WithStyles(ForwardRef(Avatar))
      alt="test"
      src="string"
    >
      <ForwardRef(Avatar)
        alt="test"
        classes={
          Object {
            "circle": "MuiAvatar-circle",
            "colorDefault": "MuiAvatar-colorDefault",
            "fallback": "MuiAvatar-fallback",
            "img": "MuiAvatar-img",
            "root": "MuiAvatar-root",
            "rounded": "MuiAvatar-rounded",
            "square": "MuiAvatar-square",
          }
        }
        src="string"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="test"
            className="MuiAvatar-img"
            src="string"
          />
        </div>
      </ForwardRef(Avatar)>
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(ListItem))
      style={
        Object {
          "backgroundColor": "#FBFBFB",
          "borderColor": "#d3d3d3",
          "borderRadius": "20px",
          "borderStyle": "solid",
          "borderTopLeftRadius": 0,
          "borderWidth": "1px",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "width": "75%",
        }
      }
    >
      <ForwardRef(ListItem)
        classes={
          Object {
            "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
            "button": "MuiListItem-button",
            "container": "MuiListItem-container",
            "dense": "MuiListItem-dense",
            "disabled": "Mui-disabled",
            "divider": "MuiListItem-divider",
            "focusVisible": "Mui-focusVisible",
            "gutters": "MuiListItem-gutters",
            "root": "MuiListItem-root",
            "secondaryAction": "MuiListItem-secondaryAction",
            "selected": "Mui-selected",
          }
        }
        style={
          Object {
            "backgroundColor": "#FBFBFB",
            "borderColor": "#d3d3d3",
            "borderRadius": "20px",
            "borderStyle": "solid",
            "borderTopLeftRadius": 0,
            "borderWidth": "1px",
            "marginBottom": "2%",
            "marginLeft": "1%",
            "width": "75%",
          }
        }
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#d3d3d3",
              "borderRadius": "20px",
              "borderStyle": "solid",
              "borderTopLeftRadius": 0,
              "borderWidth": "1px",
              "marginBottom": "2%",
              "marginLeft": "1%",
              "width": "75%",
            }
          }
        >
          <span
            key=".0"
          >
            <b
              className="user-name"
            >
              test
            </b>
             
            <br />
            <p
              className="user-tweet"
              style={
                Object {
                  "fontSize": "1em",
                  "marginRight": "5px",
                }
              }
            />
            <p
              className="created-at"
              style={
                Object {
                  "fontSize": "0.8em",
                }
              }
            >
              a few seconds ago
            </p>
          </span>
        </li>
      </ForwardRef(ListItem)>
    </WithStyles(ForwardRef(ListItem))>
  </div>
</ChatItem>
`;
